server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -1064,9 +1062,8 @@
             sc.setParameters("keyPairName", keyPairName);
         }
 
-        if (cmd instanceof ListVMsCmdByAdmin) {
-            ListVMsCmdByAdmin aCmd = (ListVMsCmdByAdmin)cmd;
-            if (aCmd.getPodId() != null) {
+        if (_accountMgr.isRootAdmin(caller.getId())) {
```

Comment:
@rhtyd Is there a better way to do this? If not, can entirely remove ListVmsCmdByAdmin since it's no longer used

Reply:
this LGTM